cpp/src/slice2matlab/Main.cpp

```diff
@@ -1744,7 +1752,19 @@ CodeVisitor::visitClassDefStart(const ClassDefPtr& p)
             //
             // Constructor
             //
-            if(!allMembers.empty())
+            if(allMembers.empty())
+            {
+                out << nl << "function " << self << " = " << name << spar << "noInit" << epar;
+                out.inc();
+                out << nl << "if nargin == 1 && ~strcmp(noInit, 'NoInit')";
```

Comment:
The problem is that if you have a class with a single string member, you cannot assign "NoInit" to the data member using the constructor,  I guess there is little chance for this to happen, but is kind of hacky.

Reply:
Right. I don't remember why I used a string there but I've pushed a fix to use a `IceInternal.NoInit.Instance` constant instead.